vllm/worker/cpu_model_runner.py

```diff
@@ -104,65 +104,289 @@ def from_broadcasted_tensor_dict(
+                elif seq_len <= prefix_cache_len:
+                    # Full hit. Only compute the last token to avoid
+                    # erroneous behavior. FIXME: Ideally we should directly
+                    # mark all tokens as computed in the scheduler and do not
+                    # schedule this sequence, so this case should not happen.
+                    context_len = seq_len - 1
+                    token_chunk_size = 1
+
+            tokens = seq_data.get_token_ids()
+            tokens = tokens[context_len:seq_len]
+            token_positions = range(context_len, seq_len)
+
+            # For encoder-only models, the block_table is None,
+            # and there is no need to initialize the slot_mapping.
+            if block_table is not None:
+                slot_mapping = [_PAD_SLOT_ID] * len(token_positions)
+                for i, pos in enumerate(token_positions):
+                    block_number = block_table[pos // block_size]
+                    block_offset = pos % block_size
+                    slot = block_number * block_size + block_offset
+                    slot_mapping[i] = slot
+                data.slot_mapping.extend(slot_mapping)
+
+            # The MRPOE positions are prepared in _compute_multi_modal_input
+            if data.input_positions is not None:
+                data.input_positions.extend(token_positions)
+
+            # Update fields
+            data.input_tokens.extend(tokens)
+            data.num_prefills += 1
+            data.num_prefill_tokens += len(tokens)
+            data.query_lens.append(len(tokens))
+            data.prefill_block_tables.append(block_table)
+        else:
+            tokens = seq_data.get_last_token_id()
+            token_positions = seq_len - 1
+            block_number = block_table[token_positions // block_size]
+            block_offset = token_positions % block_size
+            slot = block_number * block_size + block_offset
+
+            # For paged_attention kernel
+            if self.runner.sliding_window:
+                start_idx = max(0, seq_len - self.runner.sliding_window)
+                start_block = start_idx // block_size
+                start_idx = start_block * block_size
+                seq_len = seq_len - start_idx
+                block_table = block_table[start_block:]
+
+            # For MRotaryEmbedding
+            if data.input_positions is None:
+                next_pos = MRotaryEmbedding.get_next_input_positions(
+                    seq_data.mrope_position_delta,
+                    context_len,
+                    seq_len,
+                )
+                data.input_mrope_positions[0].extend(  # type: ignore
+                    next_pos[0])
+                data.input_mrope_positions[1].extend(  # type: ignore
+                    next_pos[1])
+                data.input_mrope_positions[2].extend(  # type: ignore 
+                    next_pos[2])
```

Comment:
```suggestion
                for idx in range(3):
                        input_mrope_positions[idx].extend(next_pos[idx])
```
I think we can keep the previous for loop here to avoid duplicate code.